### ISSUE TITLE
Ensure crate abides by the Unicode Data Files and Software License

### ISSUE THREAD

**Issue by lopopolo** [BurntSushi/bstr#129]
In practice I believe this means adding `AND Unicode-DFS-2016` to the SPDX license expression.

See:

- https://github.com/artichoke/focaccia/pull/149
- https://github.com/dtolnay/unicode-ident/pull/4#issuecomment-1158277839

**Comment by lopopolo** [comment]
cc @dtolnay who first uncovered this with some Google folks.

**Comment by BurntSushi** [comment]
I'm fine to add license notices and files, but I will not be re-licensing bstr itself. I've already done this dance with Google people for the regex crate: https://github.com/rust-lang/regex/issues/530

In essence, I brought the Unicode license notice in and added a note about it to the `regex-syntax` README. I do not think anything else is legally required.

The only way I'd be willing to consider changing the crate license itself is if the Rust Project itself did so. The standard library, for example, makes use of data from the Unicode Character Database. Yet, the standard library is licensed as MIT/Apache-2.0. And I can't even find any Unicode license notices at all: https://github.com/rust-lang/rust/tree/master/library/core/src/unicode

Basically, I don't see any material difference in how bstr uses the UCD versus how std uses the UCD. So I won't even consider changing the license of this crate until std does. (Or unless someone makes a compelling argument for why std is somehow different than bstr.)

Otherwise, if someone cares enough about this, I'm happy to accept a PR like [the one for `regex-syntax`](https://github.com/rust-lang/regex/pull/535/files) that adds the Unicode license and a short notice in the README.

**Comment by BurntSushi** [comment]
For anyone following this, see: https://github.com/rust-lang/rust/issues/98116#issuecomment-1359471815

**Comment by BurntSushi** [comment]
Given https://github.com/rust-lang/rust/issues/98116#issuecomment-1359471815 and my own general inclination against  relicensing the entire crate just because we use some Unicode data. I don't mean that in a "I just don't want to do it" sense, but in a "it's not at all clear that it is required in order to be in compliance with the Unicode consortium's license."

There's also some minor risk analysis here: the Unicode license is permissive. Even if we did get this wrong in some way, the actual practical fallout for it is unlikely to ever materialize.
